partiql-grammar/src/main/antlr/org/partiql/grammar/PartiQL.g4

```diff
@@ -0,0 +1,614 @@
+    | FULL OUTER?
+    | OUTER
+    ;
+
+/**
+ *
+ * EXPRESSIONS & PRECEDENCE
+ *
+ * Precedence Table:
+ * 1. Primary Expressions: Functions, Literals, Paths, Identifiers, etc (ex: a, f(a), 1, a.b, "a")
+ * 2. Unary plus, minus (ex: -a, +a)
+ * 3. Multiplication, Division, Modulo (ex: a * b)
+ * 4. Addition, Subtraction (ex: a + b)
+ * 5. Other operators (ex: a || b)
+ * 6. Predicates (ex: a LIKE b, a < b, a IN b, a = b)
+ * 7. IS true/false. Not yet implemented in PartiQL, but defined in SQL-92. (ex: a IS TRUE)
+ * 8. NOT (ex: NOT a)
+ * 8. AND (ex: a AND b)
+ * 9. OR (ex: a OR b)
+ *
+ */
+
+expr
+    : exprOr;
+
+exprOr
+    : lhs=exprOr OR rhs=exprAnd     # Or
+    | parent=exprAnd                # ExprOrBase
+    ;
+
+exprAnd
+    : lhs=exprAnd op=AND rhs=exprNot  # And
+    | parent=exprNot                  # ExprAndBase
+    ;
+
+exprNot
+    : <assoc=right> op=NOT rhs=exprNot  # Not
+    | parent=exprPredicate              # ExprNotBase
+    ;
+
+exprPredicate
+    : lhs=exprPredicate op=(LT_EQ|GT_EQ|ANGLE_LEFT|ANGLE_RIGHT|NEQ|EQ) rhs=mathOp00  # PredicateComparison
+    | lhs=exprPredicate IS NOT? type                                                 # PredicateIs
+    | lhs=exprPredicate NOT? IN PAREN_LEFT expr PAREN_RIGHT                          # PredicateIn
+    | lhs=exprPredicate NOT? IN rhs=mathOp00                                         # PredicateIn
+    | lhs=exprPredicate NOT? LIKE rhs=mathOp00 ( ESCAPE escape=expr )?               # PredicateLike
+    | lhs=exprPredicate NOT? BETWEEN lower=mathOp00 AND upper=mathOp00               # PredicateBetween
+    | parent=mathOp00                                                                # PredicateBase
+    ;
+
+mathOp00
+    : lhs=mathOp00 op=CONCAT rhs=mathOp01
+    | parent=mathOp01
+    ;
+
+mathOp01
+    : lhs=mathOp01 op=(PLUS|MINUS) rhs=mathOp02
+    | parent=mathOp02
+    ;
+
+mathOp02
+    : lhs=mathOp02 op=(PERCENT|ASTERISK|SLASH_FORWARD) rhs=valueExpr
+    | parent=valueExpr
+    ;
+
+valueExpr
+    : sign=(PLUS|MINUS) rhs=valueExpr
+    | parent=exprPrimary
+    ;
+
+exprPrimary
+    : cast                       # ExprPrimaryBase
+    | sequenceConstructor        # ExprPrimaryBase
+    | substring                  # ExprPrimaryBase
+    | canCast                    # ExprPrimaryBase
+    | canLosslessCast            # ExprPrimaryBase
+    | extract                    # ExprPrimaryBase
+    | coalesce                   # ExprPrimaryBase
+    | dateFunction               # ExprPrimaryBase
+    | aggregate                  # ExprPrimaryBase
+    | trimFunction               # ExprPrimaryBase
+    | functionCall               # ExprPrimaryBase
+    | nullIf                     # ExprPrimaryBase
+    | exprPrimary pathStep+      # ExprPrimaryPath
+    | caseExpr                   # ExprPrimaryBase
+    | valueList                  # ExprPrimaryBase
+    | values                     # ExprPrimaryBase
+    | exprTerm                   # ExprPrimaryBase
```

Comment:
You do not need the alternative labels here

Reply:
Note the ExprPrimaryPath alternative label